### DISCUSSION TITLE
It is possible to get Face value from a combobox widget?

### DISCUSSION THREAD

**Discussion by australgisinc** [pymupdf/PyMuPDF#2697] [Looking for help] [answered]
I am trying to extract a chosen value from a combobox widget using the following code:

#Extract data from combobox
import fitz
fileIN_Master = "Mypdf.pdf"     
with fitz.open(fileIN_Master) as doc:
for page in doc: 
widgets = page.widgets()
for widget in widgets:
if widget.field_type_string in ('ComboBox'):
print('field_name:', widget.field_name, 'field_value:', widget.field_value)

All I can get is field name and export value. I was wondering if it is possible to get the face value as well.

**Comment by JorjMcKie** [comment]
You can access all the choice values, find the field value in that list and the face value from there.

**Reply by australgisinc** [reply]
Thanks for your quick answer,  but I don't understand. May you please, be more specific? What list are you talking about? Thanks!

**Reply by JorjMcKie** [reply, marked as answer]
Ok, sorry. Please look at this example. We are looking at a combo and would like to see both values based on the selected (field) value:
```python
In [1]: import fitz
In [2]: doc = fitz.open("widgettest-adobe.pdf")
In [3]: page = doc[0]
In [4]: w = page.load_widget(32)  # please ignore: we happen to know th xref in this case
In [5]: w.field_value  # a combo box, let's look at its selected value
Out[5]: 'England'
In [6]: [item for item in w.choice_values if w.field_value in item]  # see the complete item
Out[6]: [('England', 'GB')]
In [7]:
```

**Reply by JorjMcKie** [reply]
You are aware presumably, that a combo box item may consist of 1 or 2 items.
```python
In [8]: w.choice_values
Out[8]:
[('Spanien', 'ES'),
 ('Frankreich', 'F'),
 ('Holland', 'NL'),
 ('Dänemark', 'DK'),
 ('Schweden', 'S'),
 ('Norwegen', 'N'),
 ('England', 'GB'),
 ('Polen', 'P'),
 'Italien',
 'Portugal',
 'Griechenland']
In [9]:
```

**Reply by australgisinc** [reply]
YES! that's a fantastic idea. It didn't come to my mind using widget.choice_values in that way. 
Thank you very much!
